tests-beta/ios/MASVS-CRYPTO/MASTG-TEST-0304.md

```diff
@@ -0,0 +1,69 @@
+
+```c
+CCCryptorStatus CCCrypt(
+    CCOperation op,         /* kCCEncrypt, etc. */
+    CCAlgorithm alg,        /* kCCAlgorithmAES128, etc. */
+    CCOptions options,      /* kCCOptionPKCS7Padding, kCCOptionECBMode, etc. */
+    const void *key,
+    size_t keyLength,
+    const void *iv,         /* optional initialization vector */
+    const void *dataIn,
+    size_t dataInLength,
+    void *dataOut,
+    size_t dataOutAvailable,
+    size_t *dataOutMoved);
+```
+
+The `CCOptions` parameter can include `kCCOptionECBMode` (value `0x0002` or `2`) to enable ECB mode. For example:
+
+```swift
+CCCrypt(
+    CCOperation(kCCEncrypt),
+    CCAlgorithm(kCCAlgorithmAES),
+    CCOptions(kCCOptionECBMode),  // ECB mode enabled
+    ...
+)
+```
+
+In this test, we're going to focus on symmetric encryption modes such as [ECB (Electronic Codebook)](https://en.wikipedia.org/wiki/Block_cipher_mode_of_operation#Electronic_codebook_(ECB)).
+
+ECB (defined in [NIST SP 800-38A](https://csrc.nist.gov/pubs/sp/800/38/a/final)) is generally discouraged [see NIST announcement in 2023](https://csrc.nist.gov/news/2023/decision-to-revise-nist-sp-800-38a) due to its inherent security weaknesses. While not explicitly prohibited, its use is limited and advised against in most scenarios. ECB is a block cipher mode that operates deterministically, dividing plaintext into blocks and encrypting them separately, which reveals patterns in the ciphertext. This makes it vulnerable to attacks like [known-plaintext attacks](https://en.wikipedia.org/wiki/Known-plaintext_attack) and [chosen-plaintext attacks](https://en.wikipedia.org/wiki/Chosen-plaintext_attack).
+
+When `kCCOptionECBMode` is set in the options parameter, the encryption uses ECB mode, which is considered vulnerable. The default behavior (when `kCCOptionECBMode` is not set) is to use CBC mode, which is more secure when used with a proper initialization vector (IV).
+
+You can learn more about ECB and other modes in [NIST SP 800-38A - Recommendation for Block Cipher Modes of Operation: Methods and Techniques](https://csrc.nist.gov/pubs/sp/800/38/a/final). Also check the [Decision to Revise NIST SP 800-38A, Recommendation for Block Cipher Modes of Operation: Methods and Techniques](https://csrc.nist.gov/news/2023/decision-to-revise-nist-sp-800-38a) and [NIST IR 8459 Report on the Block Cipher Modes of Operation in the NIST SP 800-38 Series](https://nvlpubs.nist.gov/nistpubs/ir/2024/NIST.IR.8459.pdf) for the latest information.
+
+**Note**: CryptoKit does not support ECB mode and is therefore not vulnerable to this issue. CryptoKit only supports secure encryption modes like AES-GCM and ChaCha20-Poly1305.
```

Comment:
What if the app uses an 3rd-party lib?

Reply:
![window-](https://github.com/user-attachments/assets/48466176-e640-45a7-80ce-fb931de988a8)